Mathlib/Tactic/Linter/Lint.lean

```diff
@@ -218,6 +218,56 @@ initialize addLinter cdotLinter
 
 end CDotLinter
 
+/-!
+#  The "longFile" linter
+
+The "longFile" linter emits a warning on files whose line number exceeds the value of the linter's
+option.
+The default value for the option is `1500`.
+-/
+
+/--
+The "longFile" linter emits a warning on files whose line number exceeds the value of the linter's
+option.
+The default value for the option is `1500`.
+A value of `0` silences the linter entirely.
+-/
+register_option linter.longFile : Nat := {
+  defValue := 1500
+  descr := "enable the longFile linter"
+}
+
+namespace LongFile
+
+@[inherit_doc Mathlib.Linter.linter.longFile]
+def longFileLinter : Linter where run := withSetOptionIn fun stx ↦ do
+  let linterBound := linter.longFile.get (← getOptions)
+  if linterBound == 0 then
+    return
+  unless stx.isOfKind ``Lean.Parser.Command.eoi do return
+  if (← getMainModule) == `Mathlib then return
+  if let some init := stx.getPos? then
+    -- the last line: we subtract 1, since the last line is expected to be empty
+    let lastLine := ((← getFileMap).toPosition init).line - 1
+    -- `candidate` is divisible by `100` and satisfies `lastLine + 100 < candidate ≤ lastLine + 200`
+    let candidate := (lastLine / 100) * 100 + 200
+    let endMsg := m!"using `set_option linter.longFile {candidate}`.\n\
+                    You can disable completely this linter by setting the length limit to `0`."
+    if linterBound < lastLine then
+      logWarningAt stx <| .tagged linter.longFile.name
+        m!"This file is {lastLine} lines long, but the limit is {linterBound}.\n\n\
+          You can extend the default length of the file {endMsg}"
+    else
+    let candidate := max candidate 1500
```

Comment:
I disallowed forcing "short" files.  I agree that currently no file is required to be short, so allowing to set a value of less than 1500 is not really justified.

Reply:
That is reasonable, good idea!

I took a detailed look at this logic again: it would still *suggest* a new limit shorter than the default.
At the same time, the equality case also needs to be handled: if a file has 300 lines, but a limit of 2000, something should get printed --- ideally, a suggestion to remove the set_option. I added both.

Yes, this code is more subtle than I would like...

Reply:
I think that this is fixed by the current iteration of the linter.

Reply:
So did I!